CoffeeChat/Controller/EditingViewController.swift

```diff
@@ -233,15 +254,76 @@ class EditingViewController: UIViewController {
         navigationItem.rightBarButtonItem = saveBarButtonItem
     }
 
+    // MARK: - Button Action
     @objc private func backPressed() {
         navigationController?.popViewController(animated: true)
     }
 
     @objc func savePressed() {
-        // TODO save favorites
+        if let yourSection = yourSection {
+            var interests: [Interest] = []
+            var groups: [Group] = []
+
+            for item in yourSection.items {
+                switch item {
+                case .interest(let interest):
+                    interests.append(interest)
+                case .group(let group):
+                    groups.append(group)
+                }
+            }
+
+            if interests.count > 0 {
+                NetworkManager.shared.updateUserInterests(interests: interests.map({ $0.name })).observe { result in
```

Comment:
I think you can use key paths here and say `... interests.map(\.name)).observe ...`